Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -127,36 +130,89 @@ public class PushDeploymentController : ApiController
             }
         }
 
+        private async Task<string> GetZipURLFromJSON()
+        {
+            // ARM template should have properties field and a packageUri field inside the properties field.
+            var requestObject = await Request.Content.ReadAsAsync<JObject>();
+            using (_tracer.Step("Reading the zip URL from the request JSON"))
+            {
+                try
+                {
+                    return requestObject.Value<JObject>("properties").Value<string>("packageUri");
+                }
+                catch (Exception ex)
+                {
+                    _tracer.TraceError(ex, "Error reading the URL from the JSON {0}", requestObject.ToString());
+                    throw ex;
+                }
+            }
+        }
+
         private async Task<HttpResponseMessage> PushDeployAsync(ZipDeploymentInfo deploymentInfo, bool isAsync)
         {
-            if (_settings.RunFromLocalZip())
+            var content = Request.Content;
+            var isRequestJSON = content.Headers.ContentType.MediaType.Equals("application/json", StringComparison.OrdinalIgnoreCase);
+            if (isRequestJSON)
             {
-                await WriteSitePackageZip(deploymentInfo, _tracer);
+                try
+                {
+                    deploymentInfo.ZipURL = await GetZipURLFromJSON();
+                }
+                catch (Exception)
+                {
+                    var badResponse = Request.CreateResponse();
+                    badResponse.StatusCode = HttpStatusCode.BadRequest;
+                    return badResponse;
```

Comment:
Won't that cause a generic 400 that gives no indication about the missing URL in the json?

Reply:
@davidebbo I can update to send an error message back in JSON. I am thinking of adding a check if it's an ARM request then we can send an ARM error response. But, if it's a generic JSON request, we send a simple JSON with the error.